src/features/Admin/Ads/AddNewAd/AddNewAd.tsx

```diff
@@ -0,0 +1,186 @@
+                navigate('/home/ads');
+                toast.success("Ads Add Successfully");
+            })
+            .catch((error) => {
+                toast.error("Error Occurred")
+            })
+            .finally(() => setIsLoading(false));
+    };
+
+    // Get All Rooms
+    const getAllRooms = () => {
+        axios
+            .get(`${roomsUrl}`, {
+                headers: requestHeaders,
+            })
+            .then((response) => {
+                setRoomsList(response.data.data.rooms);
+            })
+            .catch((error) => { });
+    };
+
+    useEffect(() => {
+        getAllRooms();
+    }, []);
+
+    return (
+        <>
+            <Container component="main" className={style.wrapper} sx={{ width: '100%', padding: 0 }}>
+                <Grid container justifyContent="center">
+                    <Grid item xs={12} sm={12} md={12} component={Paper} elevation={6} mb={8} mt={4} sx={{ padding: '2rem' }}>
+                        <Box
+                            sx={{
+                                my: 4,
+                                mx: 'auto',
+                                display: 'flex',
+                                flexDirection: 'column',
+                                width: '100%',
+                                maxWidth: '100%',
+                            }}
+                        >
+                            <form noValidate onSubmit={handleSubmit(onSubmit)}>
+                                <FormControl fullWidth sx={{ marginBottom: '1rem' }}>
+                                    <InputLabel id="demo-simple-select-label">Room Number</InputLabel>
+                                    <Select
+                                        {...register('room', { required: true })}
+                                        labelId="demo-simple-select-label"
+                                        id="demo-simple-select"
+                                        value={getValues('room') || ''}
+                                        label="Room Number"
+                                        onChange={(e) => setValue('room', e.target.value || '', { shouldValidate: true })}
+                                    >
+                                        {roomsList?.map((room) => (
+                                            <MenuItem key={room._id} value={room._id}>
+                                                {room.roomNumber}
+                                            </MenuItem>
+                                        ))}
+                                    </Select>
+                                </FormControl>
+                                {errors.room && <span className="errorMsg">Room is required</span>}
+
+                                <FormControl fullWidth sx={{ marginBottom: '1rem' }}>
+                                    <InputLabel id="demo-simple-select-label">Active</InputLabel>
+                                    <Select
```

Comment:
This is optional. We can simplify by setting `data.isActive = true` directly in the onSubmit function. This assumes that, in most cases, new ads will be active by default, unless there's a need to specify a activation period due to API changes.